core/src/binary_data.rs

```diff
@@ -0,0 +1,30 @@
+use crate::context::UpdateContext;
+use gc_arena::{Collect, GcCell};
+
+#[derive(Debug, Clone, Collect, Copy)]
+#[collect(no_drop)]
+pub struct BinaryData<'gc>(GcCell<'gc, UnderlyingBinaryData>);
+
+#[derive(Debug, Clone, Collect)]
+#[collect(require_static)]
+struct UnderlyingBinaryData {
+    pub data: Vec<u8>,
```

Comment:
feels like we have too many copies here. Ideally constructing the ByteArray should do one copy from the SWF to the memory, I think?
So `UnderlyingBinaryData` should store a reference to SWF's slice (possibly an SwfSlice?), and `data()` should also return a slice - then only `byte_array.write_bytes()` will do actual copying.

Reply:
I kinda started doing this but it wanted an explicit life time on the slice/vector and it started propagating througout the code to Character and beyond and I felt I was doing something wrong. I'm not that comfortable with rust yet.
If you have guidance, it would be great. Or alternatively, open an issue/add to tracking while accepting this pull request.

Reply:
Hmm, my understanding was that SwfSlice is such an escape hatch, it doesn't depend on any lifetimes as it just stores an `Arc<SwfMovie>` and numeric indices.